dvc/cli.py

```diff
@@ -204,6 +237,8 @@ def get_main_parser():
     for cmd in COMMANDS:
         cmd.add_parser(subparsers, parent_parser)
 
+    parser.cmd_choices = list(subparsers.choices.keys())
```

Comment:
It looks a bit strange to be setting this here, and my first thought was that `DvcParser.cmd_choices` should be a function or property that can dynamically figure out what the command choices are.

But for the scope of this PR, I think this is fine. (Unless you really feel like digging into `argparse.ArgumentParser` internals 😄 )

Reply:
Yea it does feel a bit strange but this appears to be the only place we have access to the `choices` field. I can dig into the internals some more and see if we can get this elsewhere.

Reply:
I took a quick look at the argparse code as well, and I think it's probably possible to do something with the `ArgumentParser._subparsers` property, but nothing stood out as a quick and obvious solution to me. What you have now should be fine for our purposes, so don't worry about spending too much time digging into this one.